concordium-consensus/src/Concordium/KonsensusV1/Types.hs

```diff
@@ -365,6 +366,18 @@ checkQuorumCertificate qsmGenesis sigThreshold FinalizationCommittee{..} QuorumC
             Nothing -> False
             Just FinalizerInfo{..} -> check (finalizerWeight + accumWeight) (finalizerBlsKey : keys) is
 
+-- |Determine the 'BakerId's of the finalizers that signed a 'QuorumCertificate'.
+-- This assumes that all of the signatories are valid indexes into the finalization committee
+-- (which is checked by 'checkQuorumCertificate'). Any invalid finalization indexes will be
+-- omitted.
```

Comment:
Maybe we should document the order of the list returned, I suppose their ordered by ascending baker id.

And a single fold is sufficient since the finalizers are in ascending order based on their respective baker id.